gp-project-contributors.php

```diff
@@ -112,6 +113,132 @@ public function gp_project_contributors( $atts ) {
+		$gpl = new GP_Locales;
+		
+		// Setup some variables to use later.
+		$return = '<style type="text/css">.gptl-twitter, .gptl-twitter:focus, .gptl-twitter:hover, .gptl-twitter:link, .gptl-twitter:visited, .gptl-twitter:active { color: #55acee; } .gptl-facebook, .gptl-facebook:focus, .gptl-facebook:hover, .gptl-facebook:link, .gptl-facebook:visited, .gptl-facebook:active { color: #3A5795; }</style><table style="border: 0px;">';
+		$names = array();
+        	$links = array();
+		$languageContributions = array();
+		
+		// Grab all of the contributors from the GlotPress translations table and join it to the WordPress users table so we can get display names later.
+		$result = $wpdb->get_results( "SELECT {$wpdb->users}.*, 
+SUM(CASE {$gp_table_prefix}translations.status WHEN 'current' THEN 1 ELSE 0 END) current_contrib, 
+SUM(CASE {$gp_table_prefix}translations.status WHEN 'fuzzy' THEN 1 ELSE 0 END) fuzzy_contrib, 
+SUM(CASE {$gp_table_prefix}translations.status WHEN 'waiting' THEN 1 ELSE 0 END) waiting_contrib, 
+SUM(CASE {$gp_table_prefix}translations.status WHEN 'old' THEN 1 ELSE 0 END) old_contrib, 
+SUM(CASE {$gp_table_prefix}translations.status WHEN 'rejected' THEN 1 ELSE 0 END) rejected_contrib, 
+COUNT(DISTINCT {$gp_table_prefix}translations.id) total_contrib, 
+{$gp_table_prefix}translation_sets.*
+FROM {$gp_table_prefix}translations
+INNER JOIN {$gp_table_prefix}translation_sets ON {$gp_table_prefix}translations.translation_set_id = {$gp_table_prefix}translation_sets.id
+INNER JOIN `{$wpdb->users}` on `{$gp_table_prefix}translations`.`user_id` = `{$wpdb->users}`.`ID`
+INNER JOIN {$gp_table_prefix}originals ON {$gp_table_prefix}translations.original_id = {$gp_table_prefix}originals.id
+WHERE `{$gp_table_prefix}originals`.`project_id` = {$project_id}
+AND `{$gp_table_prefix}originals`.`status` = '+active'
+GROUP BY {$wpdb->users}.user_login, {$gp_table_prefix}translation_sets.id" );
+		
+		// $return .= '<pre>' . print_r( $result, true ) . '</pre>';
+
+		// Loop through all the results from the database and create a list of locales with all their approvers associated with them.
+		foreach( $result as $row ) {
+			$current = $gpl->locales[$row->locale];
+			
+			$names[$current->english_name][] = $row->display_name;
+           		$links[$row->display_name] = $row->user_url;
+			
+			$contribs = new stdClass;
+			$contribs->current = $row->current_contrib;
+			$contribs->fuzzy = $row->fuzzy_contrib;
+			$contribs->waiting = $row->waiting_contrib;
+			$contribs->old = $row->old_contrib;
+			$contribs->rejected = $row->rejected_contrib;
+			
+			$contribs->total = $row->total_contrib;
+			$contribs->tooltip = "Current: " . number_format_i18n($contribs->current, 0) . "\r\n" .
+				"Fuzzy: " . number_format_i18n($contribs->fuzzy, 0) . "\r\n" .
+				"Waiting: " . number_format_i18n($contribs->waiting, 0) . "\r\n" .
+				"Old: " . number_format_i18n($contribs->old, 0) . "\r\n" .
+				"Rejected: " . number_format_i18n($contribs->rejected, 0);
```

Comment:
The Fuzzy/Waiting/Old/Rejected strings should be translated with `__()` instead of hard coded.

Reply:
Added i18n-code to the items